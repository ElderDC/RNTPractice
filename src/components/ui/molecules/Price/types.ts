export interface PriceProps {
	currency?: string;
	amount?: number;
	small?: boolean;
}
