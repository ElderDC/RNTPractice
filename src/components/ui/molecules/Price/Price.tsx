import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, View } from 'react-native';
import { useThemeAwareObject } from 'hooks/useThemeHook';
import { Theme } from 'theme/Theme.interface';
import { PriceProps } from './types';

const Price = (props: PriceProps): JSX.Element | null => {
	const { currency, amount, small } = props;

	const proccessAmount = (amount: number) => {
		const [integer, decimal] = amount.toFixed(2).split('.');
		return {
			integer,
			decimal: decimal ? `.${decimal}` : '.00',
		};
	};

	const createStyles = (theme: Theme) => {
		const styles = StyleSheet.create({
			price: {
				...theme.text[small ? 'title2' : 'h5'],
				color: theme.color.surface2,
			},
			priceDecimal: {
				...theme.text.title2,
				color: theme.color.surface2,
			},
		});
		return styles;
	};

	const styles = useThemeAwareObject(createStyles);

	return (
		<View>
			<Text style={styles.price}>
				{currency}
				{proccessAmount(amount || 0).integer}
				<Text style={styles.priceDecimal}>
					{proccessAmount(amount || 0).decimal}
				</Text>
			</Text>
		</View>
	);
};

Price.defaultProps = {
	currency: '$',
	amount: 0,
	small: false,
};

Price.propTypes = {
	currency: PropTypes.string,
	amount: PropTypes.number,
	small: PropTypes.bool,
	style: PropTypes.oneOfType([
		PropTypes.object,
		PropTypes.arrayOf(PropTypes.object),
	]),
};

export default Price;
