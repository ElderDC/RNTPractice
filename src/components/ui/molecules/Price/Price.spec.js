import React from 'react';
import renderer from 'react-test-renderer';
import Price from './Price';

describe('<Price />', () => {
	it('renders correctly', () => {
		const tree = renderer.create(<Price />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('has 1 child', () => {
		const tree = renderer.create(<Price />).toJSON();
		expect(tree.children).toHaveLength(1);
	});
});
