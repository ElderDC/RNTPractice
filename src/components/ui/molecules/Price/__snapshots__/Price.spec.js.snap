// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Price /> renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#6B7280",
        "fontSize": 32,
        "fontWeight": "700",
      }
    }
  >
    $
    0
    <Text
      style={
        Object {
          "color": "#6B7280",
          "fontSize": 20,
          "fontWeight": "700",
        }
      }
    >
      .00
    </Text>
  </Text>
</View>
`;

exports[`renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#6B7280",
        "fontSize": 32,
        "fontWeight": "700",
      }
    }
  >
    $
    0
    <Text
      style={
        Object {
          "color": "#6B7280",
          "fontSize": 20,
          "fontWeight": "700",
        }
      }
    >
      .00
    </Text>
  </Text>
</View>
`;
